File: packages/canvas-polyfill/src-native/AsyncDemo/app/src/main/java/com/github/triniwiz/asyncdemo/MainActivity.java
package com.github.triniwiz.asyncdemo;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import com.github.triniwiz.async.Async;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.File;
import java.io.UnsupportedEncodingException;

public class MainActivity extends AppCompatActivity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		File font = new File(getFilesDir(), "helvetiker_regular.typeface.json");
		Async.FileManager.readFile(font.getAbsolutePath(), null, new Async.FileManager.Callback() {
			@Override
			public void onError(String error, Exception e) {
				Log.d("com.test", "readFile error: " + error);
			}

			@Override
			public void onComplete(@Nullable Object object) {
				Log.d("com.test", "readFile success: ");
				byte[] data = (byte[]) object;
				try {
					String string = new String(data, java.nio.charset.Charset.forName("UTF-8")).toString();
					JSONObject json = new JSONObject(string);
					Log.d("com.test", "json: " + json.toString().contains("»"));
				}catch (JSONException e) {
					e.printStackTrace();
				}
			}
		});

	}
}

File: packages/canvas-polyfill/src-native/AsyncDemo/app/src/main/java/com/github/triniwiz/asyncopdemo/MainActivity.kt
package com.github.triniwiz.asyncopdemo

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.github.triniwiz.async.Async2
import org.json.JSONException
import org.json.JSONObject
import java.io.File
import java.nio.charset.Charset

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val font = File(filesDir, "helvetiker_regular.typeface.json")

        Async2.FileManager.readFile(font.absolutePath, null, object : Async2.FileManager.Callback {
            override fun onError(error: String?, e: Exception?) {
                Log.d("com.test", "readFile error: $error")
            }

            override fun onComplete(file: Any) {
                Log.d("com.test", "readFile success: ")
                val data = file as ByteArray
                try {
                    val string = String(data, Charset.forName("UTF-8"))
                    println(string)
                    val json = JSONObject(string)
                    Log.d("com.test", "json: " + json.toString().contains("»"))
                } catch (e: JSONException) {
                    e.printStackTrace()
                }
            }
        })

        Async2.Base64.base64ToFile("iVBORw0KGgoAAAANSUhEUgAAAAQAAAABAQMAAADD8p2OAAAAA1BMVEX/AP804Oa6AAAACklEQVQI12NgAAAAAgAB4iG8MwAAAABJRU5ErkJggg==", filesDir.absolutePath,object: Async2.Base64.Callback {
            override fun success(response: Any?) {
                Log.d("com.test", "base64ToFile success: $response")
            }

            override fun error(error: Any?, message: String?) {
                Log.d("com.test", "base64ToFile error: $error")
            }
        })
    }
}